src/twisted/_version.py

```diff
@@ -7,5 +7,5 @@
 
 from incremental import Version
 
-__version__ = Version('Twisted', 17, 1, 0, dev=0)
+__version__ = Version('Twisted', 17, 1, 1, dev=0)
```

Comment:
The next trunk release will be 17.2.0, not 17.1.1.

Reply:
@exarkun I'm in agreement with everything you wrote in https://github.com/twisted/twisted/pull/745#discussion_r107141416

Reply:
@glyph regarding your question:

>  In what circumstance would you find yourself with a random git revision of Twisted trunk as an installable option?

I've been telling a few people to install Twisted from my branch to test out some email fixes directly from GitHub via pip: https://github.com/buildbot/buildbot/issues/3633#issuecomment-330712358

So installing from a random git version is possible.  Therefore it is important to have
the correct version in the repository at all times, even beyond a release version.

Reply:
@rodrigc What I specifically meant by "an installable option" is, "if I `pip install twisted==X.Y.Z`, when will I get a git version as a result".  To get your version from a git repo, someone would need to do `pip install git+https://.../@...`; i.e. explicitly request installation from a particular artifact, rather than a version specifier.

But, you're right of course, that the version metadata should be as accurate as possible.  My kingdom for a bzr-style revision counter :-)

I had trouble re-understanding this thread, so I went back to [the relevant spec](https://www.python.org/dev/peps/pep-0440/) and wrote out the options I think we discussed so far:

After release X.Y.0, we have:

1. X.Y.0.post0, which is wrong because tooling may rightly consider it equivalent to X.Y.0
2. X.Y.0.dev0, which is wrong because it stipulates an *earlier* version than X.Y.0 according to the standardized sorting algorithm; this causes the aforementioned entrypoints issue, because you want to be able to use the dev version of Twisted with stuff that required at-least-the-previous-version.
3. X.(Y+1).0.dev0 which is wrong because X.(Y+1) may never exist
4. X.Y.1.dev0 which is wrong because X.Y.1 will happen off a different branch, and may not contain any of the changes present in the version of `trunk` it's applied to, etc.

In conclusion:

Upon reflection, I think I didn't really understand @exarkun's complaint, I thought he was suggesting a "Y" where he was actually suggesting "Y+1", and I now agree that option 3 is by far the best option we have.  So: sorry for my pushback, push-un-backed.

(I do still wish we'd just landed the .1 version at the time and had this discussion after the fact though :).)

Reply:
Yes, Option 3 sounds fine to me.  It basically boils down to whatever @hawkowl chooses, since she's the one controlling the releases and the version numbering.